### DISCUSSION TITLE
export collection titles to bibtex keywords for multiple bibliographies?

### DISCUSSION THREAD

**Discussion by boan-anbo** [retorquere/zotero-better-bibtex#2118] [Ideas]
Hi there! I was wondering what's the best practice for this scenario:

Suppose I have the following collections, `parent-one`, and `parent-two`. 

Each have two child sections, `parent-one` has `parent-one-child-one` and `parent-one-child-two` for example.

I was hoping to export all these collections to a single `.bib`.

With `biblatex`, I can use keywords to filter the bibliography, so that I can have one bibliography for each Zotero collection, for example,

```
\printbibliography[keyword=parent-one, heading="Parent Collection One Bibliography"]
\printbibliography[keyword=parent-two, heading="Parent Collection Two Bibliography"]
```
This way, the parent collection bibliography should also print those Zotero items in its child collections.

So I was wondering how best to achieve this in the current version of BBT. 

And if there is no current solution,  is it possible to add an option to BBT to explort the collection names to the keyword field?

For this, the keyword field should contain the collection name of the item itself and its parent collection as well. So the user could filter in `biblatex` either on that child collection's name, or its parent collection's name. 

For example, a Zotero item in the second child section of the first parent section will have the keywords of `child-one; parent-one`. This way, this entry can be selected by specifying either its own collection name or its parent collection name.

Thanks for your consideration!

**Comment by retorquere** [comment]
Adding the collection an item is present in can be done with a [postscript](https://retorque.re/zotero-better-bibtex/exporting/scripting/):

```
if (Translator.BetterTeX) {
  if (item.collections) reference.add({
    name: 'keywords',
    value: (item.tags || []).concat(item.collections.map(coll => Translator.collections[coll].name)),
    enc: 'tags',
  })
}
```

note that accessing `collections` means that the cache is disabled for these exports (so they will be slower), and that if you're exporting a collection `parent-one` then `parent-one` itself will not show up -- the "root" of the export is just offered to exporters as if they lived at the top-level.

Adding the parent is sort of possible but the postscript gets more complex. It isn't always possible in the simple sense of getting "the" parent because an item can live in multiple collections and this have multiple "parent-paths".

As to having multiple collections show up in one bib file, the only way to get that is by exporting the collection they are both part of, and if no such common parent exists, the entire library. Zotero just doesn't have a facility to select multiple collections and export them. But you can of course just export two separate collections and load both in your TeX document.
